imqsauth/http.go

```diff
@@ -614,11 +630,67 @@ func httpHandlerSetGroupRoles(central *ImqsCentral, w http.ResponseWriter, r *ht
 		} else {
 			central.Central.Log.Errorf("Could not set group roles for %v: %v", groupname, err)
 			authaus.HttpSendTxt(w, http.StatusNotAcceptable, fmt.Sprintf("Could not set group roles for %v: %v", groupname, err))
+			return
 		}
 	} else {
 		central.Central.Log.Warnf("Group '%v' not found: %v", groupname, e)
 		authaus.HttpSendTxt(w, http.StatusNotFound, fmt.Sprintf("Group '%v' not found: %v", groupname, e))
+		return
 	}
+
+	broadCastGroupChange(central, r, groupname)
+}
+
+func broadCastGroupChange(central *ImqsCentral, r *httpRequest, groupname string) {
+
+	identities, err := central.Central.GetAuthenticatorIdentities()
+	if err != nil {
+		central.Central.Log.Errorf("Unable to broadcast change, unable to read identities: %v", err)
+		return
+	}
+
+	// now find all identities which belong to the group in question
+	modifiedIdentities := []string{}
+
+	for _, identity := range identities {
+		if r.token.Identity == identity {
+			// skip user that performed the request
+			continue
+		}
+		groups := getIdentityGroupIDs(central, identity)
+		var groupnamelist []string
```

Comment:
I would name 'groups' something else, like `identityGroupIds`, so that it doesn't get confused with 'groupnamelist' and 'groupname'. But speaking of renaming.. I think the entire function could be quite a bit easier to read if the group that is being changed, and all it's associated variables, are renamed to something like 'changedGroup', 'changedGroupIds', etc.
Also important - the computation of `groupids` should be moved out of the loop. Firstly, for performance. But most importantly, because having it inside the loop really hinders comprehension. If one sees code running inside the loop, you're very heavily biased towards believing that it acts on data that changes inside the loop.
The two lines that initialize groupnamelist can also be shortened into one:
```go
groupnamelist := []string{groupname}
```